src/proto/factories/callback_factory.cpp

```diff
@@ -87,9 +87,21 @@ using factory_type = lbann::generic_factory<
   std::string,
   generate_builder_type<lbann::callback_base,
                         google::protobuf::Message const&,
-                        lbann_summary*>,
+                        std::shared_ptr<lbann_summary> const&>,
   default_key_error_policy>;
 
+namespace
```

Comment:
This doesn't need to be nested in yet another anonymous namespace (it's already in one).